README.md

```diff
@@ -1,29 +1,112 @@
-# Python Plugin Template Project
+# Horreum Client Plugin
 
-## Image Building
+This plugin uploads a structured payload to a [Horreum](https://horreum.hyperfoil.io/) server.
+
+## Using the plugin
+Build the container:
+```
+docker build . -t arcaflow-plugin-horreum-client
+```
+
+Run with the provided sample input:
+```
+cat configs/sample-input.yaml | docker run -i --rm arcaflow-plugin-horreum-client -f -
+```
 
-You can change this plugin's image version tag in
-`.github/workflows/carpenter.yaml` by editing the
-`IMAGE_TAG` variable, and pushing that change to the
-branch designated in that workflow.
 
 # Autogenerated Input/Output Documentation by Arcaflow-Docsgen Below
 
 <!-- Autogenerated documentation by arcaflow-docsgen -->
-## Hello world! (`hello-world`)
+## Horreum Client (`horreum-client`)
 
-Says hello :)
+Uploads an object to the Horreum server
 
 ### Input
 
 <table><tbody>
 <tr><th>Type:</th><td><code>scope</code></td><tr><th>Root object:</th><td>InputParams</td></tr>
-<tr><th>Properties</th><td><details><summary>name (<code>string</code>)</summary>
-                <table><tbody><tr><th>Required:</th><td>Yes</td></tr><tr><th>Type:</th><td><code>string</code></td><tr><th>Minimum length:</th><td>1</td></tr></tbody></table>
+<tr><th>Properties</th><td><details><summary>data_object (<code>map[<code>string</code>, <code>
+    any</code>]</code>)</summary>
+                <table><tbody><tr><th>Name:</th><td>data object for upload</td></tr><tr><th>Description:</th><td>Data object to be uploaded to the Horreum server</td></tr><tr><th>Required:</th><td>Yes</td></tr><tr><th>Type:</th><td><code>map[<code>string</code>, <code>
+    any</code>]</code></td><tr><td colspan="2">
+    <details>
+        <summary>Key type</summary>
+        <table><tbody><tr><th>Type:</th><td><code>string</code></td></tbody></table>
+    </details>
+</td></tr>
+<tr><td colspan="2">
+    <details>
+        <summary>Value type</summary>
+        <table><tbody><tr><th>Type:</th><td><code>
+    any</code></td></tbody></table>
+    </details>
+</td></tr>
+</tbody></table>
+            </details><details><summary>horreum_keycloak_url (<code>string</code>)</summary>
+                <table><tbody><tr><th>Name:</th><td>horreum keycloak url</td></tr><tr><th>Description:</th><td>Base URL for the Horreum Keycloak server.</td></tr><tr><th>Required:</th><td>Yes</td></tr><tr><th>Type:</th><td><code>string</code></td><tr><th>Must match pattern:</th><td><code>((http|https)\:\/\/)?[a-zA-Z0-9\.\/\?\:@\-_=#]&#43;\.([a-zA-Z]){2,6}([a-zA-Z0-9\.\&amp;\/\?\:@\-_=#])*</code></td></tr></tbody></table>
```

Comment:
I'll take a look at improving the regex, but the documentation is auto-generated, so the pattern match requirement is always going to show up there.

Reply:
That's OK...some humans can read regex's (at least some of the time), and the rest just ignore them as if they were line noise (which is pretty much what they look like 😉).

Nevertheless, it would be good to improve the regex.  😁

Reply:
Implemented your suggested regex. This isn't perfect, as you indicated, but I think it is reasonable.